piff/gp_interp_2pcf.py

```diff
@@ -0,0 +1,375 @@
+        return ystar
+
+    def return_gp_predict(self,y, X1, X2, kernel, y_err=None):
+        """Compute interpolation with gaussian process for a given kernel.
+
+        :param y:  The dependent responses.  (n_samples, n_targets)
+        :param X1:  The independent covariates.  (n_samples, n_features)
+        :param X2:  The independent covariates at which to interpolate.  (n_samples, n_features)
+        :param kernel: sklearn.gaussian_process kernel.
+        :param y_err: Error of y. (n_samples, n_targets)
+        """
+
+        if y_err is None:
+            y_err = np.zeros_like(y)
+        HT = kernel.__call__(X2,Y=X1)
+        K = kernel.__call__(X1) + np.eye(len(y))*y_err**2
+        factor = (cholesky(K, overwrite_a=True, lower=False), False)
+        alpha = cho_solve(factor, y, overwrite_b=False)
+        return np.dot(HT,alpha.reshape((len(alpha),1))).T[0]
+
+    def getProperties(self, star, logger=None):
+        """Extract the appropriate properties to use as the independent variables for the
+        interpolation.
+
+        Take self.keys from star.data
+
+        :param star:    A Star instances from which to extract the properties to use.
+
+        :returns:       A np vector of these properties.
+        """
+        return np.array([star.data[key] for key in self.keys])
+
+    def initialize(self, stars, logger=None):
+        """Initialize both the interpolator to some state prefatory to any solve iterations and
+        initialize the stars for use with this interpolator.
+
+        :param stars:   A list of Star instances to interpolate between
+        :param logger:  A logger object for logging debug info. [default: None]
+        """
+        self.nparams = len(stars[0].fit.params)
+        if self.npca>0:
+            self.nparams = self.npca
+        if len(self.kernel_template)==1:
+            self.kernels = [copy.deepcopy(self.kernel_template[0]) for i in range(self.nparams)]
+        else:
+            if len(self.kernel_template)!= self.nparams or (len(self.kernel_template)!= self.npca & self.npca!=0):
+                raise ValueError("numbers of kernel provided should be 1 (same for all parameters) or " \
+                                 "equal to the number of params (%i), number kernel provided: %i"%((self.nparams,len(self.kernel_template))))
+            else:
+                self.kernels = [copy.deepcopy(ker) for ker in self.kernel_template]
+                                        
+        return stars
+
+    def solve(self, stars=None, logger=None):
+        """Set up this GPInterp object.
+
+        :param stars:    A list of Star instances to interpolate between
+        :param logger:   A logger object for logging debug info. [default: None]
+        """
+        X = np.array([self.getProperties(star) for star in stars])
+        y = np.array([star.fit.params for star in stars])
+        y_err = np.array([np.sqrt(np.diag(star.fit.params_cov)) for star in stars])
```

Comment:
not all models produce params_cov. This should probably check that stars have params_cov, and otherwise fill with zeros

Reply:
I add per default in StarFit within star.py a params_cov wich is set to None per default. Within the test it was working like that, but I guess it should be an output of each fitting method, not only lmfit.

Reply:
Yes, I agree. I put in StarFit the variance instead of the covariance, and in gsobject, I return the variance instead of the full covariance matrix. For PixelGrid, and optical_model, I set the variance for the moment to zeros.